Updated dataloader.py with torchvision v2 and added test loader

Wanghaohan123 opened the pull request:
- Replaced deprecated torchvision.transform with torchvision.transforms.v2
- Added test dataloader
- Added dataset consistency check and stratified k-fold validation

krittin-kulrat commented on maskaugfish/dataloader.py:
```diff
@@ -1,7 +1,279 @@
+    def __getitem__(self, idx: int):
+        img_path, mask_path, label = self.samples[idx]
+
+        img = Image.open(img_path).convert("RGB")
```
Use `torchvision` library to load image to automatically create a pytorch Tensor

For reference, see [this demo](https://github.com/krittin-kulrat/MaskAugFish/blob/main/data/dataset_demo.ipynb)

krittin-kulrat commented on maskaugfish/dataloader.py:
```diff
@@ -1,7 +1,279 @@
+    if args.test_root:
+        test_samples, _ = build_index(args.test_root)
+
+    train_loader, val_loader, test_loader = make_fold_loaders(
```
Try call 2 different folds and `test_loader` should use be the same

Wanghaohan123 commented:
Hi,I've refactored the dataloader based on your feedback:
- Replaced make_fold_loaders() with a new make_dataloaders() API.
- Added StratifiedShuffleSplit to create a fixed test split before K-Fold.
- Kept StratifiedKFold on the remaining pool for train/val.
- Removed redundant train/augment flags; now supports a direct augmentation pipeline (torchvision.transforms.v2.Compose).
- Switched to torchvision.io.read_image for tensor-based I/O.
- Added class frequency print + reproducibility sanity check.
- Verified locally on Fish4Knowledge (23 classes, ~27k samples): all loaders (train/val/test) return correct batch shapes [B, 3, 224, 224].

Ready for re-review 🙂

krittin-kulrat commented on maskaugfish/dataloader.py:
```diff
@@ -1,7 +1,483 @@
+        # --- load mask if exists ---
+        if mask_path is not None and Path(mask_path).exists():
+            # mask might be single-channel or rgb; take first channel after load
+            mask_raw = tvio.read_image(mask_path)  # [C,H,W] uint8
```
<img width="745" height="204" alt="image" src="https://github.com/user-attachments/assets/18472aab-e1b2-4ee0-b788-15eb850654e4" />

See [This demo](https://github.com/krittin-kulrat/MaskAugFish/blob/feature/augmentation/demo/dataset_demo.ipynb)
